process/smartContract/process.go

```diff
@@ -2004,14 +2035,16 @@ func (sc *scProcessor) createSmartContractResults(
 		outputTransferCopy := outputTransfer
 		isLastOutTransfer := i == lenOutTransfers-1
 		if !createdAsyncCallBack && isLastOutTransfer && isOutTransferTxRcvAddr &&
-			sc.useLastTransferAsAsyncCallBackWhenNeeded(callType, outAcc, &outputTransferCopy, vmOutput, tx, result) {
+			sc.useLastTransferAsAsyncCallBackWhenNeeded(callType, outAcc, &outputTransferCopy, vmOutput, tx, result, isCrossShard) {
 			createdAsyncCallBack = true
 		}
 
 		if result.CallType == vmData.AsynchronousCall {
-			result.GasLimit += outputTransfer.GasLocked
-			lastArgAsGasLocked := "@" + hex.EncodeToString(big.NewInt(0).SetUint64(outputTransfer.GasLocked).Bytes())
-			result.Data = append(result.Data, []byte(lastArgAsGasLocked)...)
+			if !sc.flagCreatedCallBackCrossShardOnly.IsSet() || isCrossShard {
```

Comment:
also here, we weren't conditioned by being cross-shard in the old version

Reply:
the gasLocked would be added and encoded to the SCR which was not needed, as intra shard async calls and callbacks were already resolved by arwen.